### PR TITLE
Refactor parts of Plane into a `plane-common` crate (DIS-2967)

### PR DESCRIPTION
**Pull request by paulgb**
The main `plane` crate contains a client, but depending on it in another project requires bringing in _all_ of Plane, including the openssl dependency. This breaks the client (and other pieces) out into a `plane-common` crate with the parts needed to "consume" Plane as a client, and a crate for Plane itself (in particular, without the database and ACME stuff, which bring in other dependencies that can be hard to wrangle).

This looks like a huge PR but there is essentially no net new code here; files are just moved around. A few notes:
- I got rid of `plane-dynamic`; it was a hack to make tests compile faster but it should be less necessary now that we have broken up Plane into several crates.
- Roughly, the criteria for "what goes in common" is "the client and everything it depends on". `plane` depends on `plane-common`, but `plane-common` does NOT depend on `plane`, so things like `ExponentialBackoff` (used by the client to manage reconnects) go in `plane-common`.

Todo:
- [x] get `test_get_metrics` and `backend_lifecycle` tests to pass.
- [x] rename to `plane-common`?
- [x] p2 PR for refactor and ensure p2 integration tests pass

### CODE REVIEW

File: plane/src/controller/core.rs [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
     names::{AnyNodeName, ControllerName},
     typed_socket::Handshake,
     types::{ClusterName, ConnectRequest, ConnectResponse, NodeId},
+    PlaneClient,
 };
-use chrono::{DateTime, Utc};
 use std::net::IpAddr;
 use url::Url;
 
+use crate::database::{connect::ConnectError, PlaneDatabase};
```
Comment (michaelsilver) [inline plane/src/controller/core.rs:11]: Nit: unintentional newline separating imports?

File: plane/src/drone/command.rs [hunk 2]
```diff
@@ -1,13 +1,16 @@
-use super::{runtime::unix_socket::UnixSocketRuntimeConfig, ExecutorConfig};
-use crate::{
-    drone::{runtime::docker::DockerRuntimeConfig, DroneConfig},
-    names::{DroneName, OrRandom},
-    types::{ClusterName, DronePoolName},
-    util::resolve_hostname,
+use crate::util::resolve_hostname;
+
```
Comment (michaelsilver) [inline plane/src/drone/command.rs:2]: Nit: unintentional newline separating imports?

File: plane/src/drone/executor.rs [hunk 3]
```diff
@@ -1,15 +1,16 @@
-use super::{backend_manager::BackendManager, state_store::StateStore};
-use crate::{
-    drone::runtime::Runtime,
-    names::BackendName,
-    protocol::{BackendAction, BackendEventId, BackendStateMessage},
-    types::{BackendState, BackendStatus, TerminationReason},
-    util::{ExponentialBackoff, GuardHandle},
-};
+use crate::util::GuardHandle;
+
```
Comment (michaelsilver) [inline plane/src/drone/executor.rs:2]: Nit: unintentional newline separating imports?

File: plane/src/drone/key_manager.rs [hunk 4]
```diff
@@ -1,13 +1,14 @@
+use crate::util::GuardHandle;
+
```
Comment (michaelsilver) [inline plane/src/drone/key_manager.rs:2]: Nit: unintentional newline separating imports?

File: plane/src/drone/mod.rs [hunk 5]
```diff
@@ -1,3 +1,5 @@
+use crate::signals::wait_for_shutdown_signal;
+
```
Comment (michaelsilver) [inline plane/src/drone/mod.rs:2]: Nit: unintentional newline separating imports?

File: plane/src/proxy/connection_monitor.rs [hunk 6]
```diff
@@ -1,11 +1,13 @@
-use crate::{heartbeat_consts::HEARTBEAT_INTERVAL, names::BackendName};
+use plane_common::names::BackendName;
 use std::{
     collections::{hash_map::Entry, HashMap, VecDeque},
     sync::{Arc, Mutex},
     time::SystemTime,
 };
 use tokio::task::JoinHandle;
```
Comment (michaelsilver) [inline plane/src/proxy/connection_monitor.rs:8]: Nit: unintentional newline separating imports?

**Review by michaelsilver — APPROVED** [review]
Some small nits about import whitespace, otherwise LGTM!

**Comment by paulgb** [comment]
> Some small nits about import whitespace, otherwise LGTM!

Fixed, thanks!

**Pull request merged**